src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -1367,7 +1367,7 @@ type IncrementalBuilder(ctokCtor: CompilationThreadToken, frameworkTcImportsCach
     let assertNotDisposed() =
         if disposed then  
             System.Diagnostics.Debug.Assert(false, "IncrementalBuild object has already been disposed!")
-    let mutable referenceCount = 0
+    let referenceCount = ref 0
```

Comment:
I think the count should start at 1. Decrementing it to 0 triggers a dispose, and incrementing it from 0 to 1 is illegal, because then the object has already been disposed in another thread..